tests/conftest.py

```diff
@@ -1,6 +1,72 @@
 from .common import *
 
 
+@pytest.fixture(scope="session")
+def adata_pbmc():
+    #adata_ref = sc.datasets.pbmc3k_processed()
+    # quick fix for broken dataset paths, should be removed with scanpy>=1.6.0
+    adata_ref = sc.read(
+        "pbmc3k_processed.h5ad",
+        backup_url="https://raw.githubusercontent.com/chanzuckerberg/cellxgene/main/example-dataset/pbmc3k.h5ad"
+    )
+    adata = sc.datasets.pbmc68k_reduced()
+
+    var_names = adata_ref.var_names.intersection(adata.var_names)
+    adata_ref = adata_ref[:, var_names]
+    adata = adata[:, var_names]
+
+    sc.pp.pca(adata_ref)
+    sc.pp.neighbors(adata_ref)
+    sc.tl.umap(adata_ref)
+
+    # merge cell type labels
+    sc.tl.ingest(adata, adata_ref, obs='louvain')
+    adata_concat = adata_ref.concatenate(adata, batch_categories=['ref', 'new'])
+    adata_concat.obs.louvain = adata_concat.obs.louvain.astype('category')
+    # fix category ordering
+    adata_concat.obs.louvain.cat.reorder_categories(adata_ref.obs.louvain.cat.categories, inplace=True)
+    adata_concat.obs['celltype'] = adata_concat.obs['louvain']
+
+    del adata_concat.obs['louvain']
+    del adata_concat.uns
+    del adata_concat.obsm
+    del adata_concat.varm
+
+    yield adata_concat
+    del adata_concat
+
+
+@pytest.fixture(scope="function")
+def adata(adata_pbmc):
+    adata = adata_pbmc.copy()
+    yield adata
+    del adata
+
+@pytest.fixture(scope="session")
+def adata_pca():
+    def adata_pca_(adata):
+        scIB.pp.reduce_data(adata, pca=True, n_top_genes=200, neighbors=False, umap=False)
+        return adata
+    return adata_pca_
+
+
+@pytest.fixture(scope="session")
+def adata_neighbors():
```

Comment:
is this intentionally not dependent on the `adata` fixture? You only ever use these together I think. It's fine to make this more general i guess, but then is order of fixtures given in the tests below?

Reply:
Never mind... you don't use these as dependent fixtures, but as separate calls.

Reply:
I guess this could have been a simple function instead of a fixture, tbh

Reply:
I changed it to a fixture